nemo_aligner/utils/distributed.py

```diff
@@ -271,27 +288,17 @@ def from_parallel_logits_to_logprobs(vocab_parallel_logits, target, inference_on
     ].contiguous()
 
 
-def pad_tensors_to_max_global_seq_len(list_of_tensors, pad_value, group, sequence_length_to_pad_to=None):
-    """pad a list of tensors to the global sequence length across the specified group
-    """
-    # compute the local padding
-    tensors_padded = torch.nn.utils.rnn.pad_sequence(list_of_tensors, batch_first=True, padding_value=pad_value)
-
-    # find global max seq length
-    max_seq_length = torch.tensor([tensors_padded.size(-1)], dtype=torch.float32, device=torch.cuda.current_device())
-    torch.distributed.all_reduce(max_seq_length, op=torch.distributed.ReduceOp.MAX, group=group)
-    max_seq_length = int(max_seq_length)
-
-    if sequence_length_to_pad_to is not None:
-        if max_seq_length > sequence_length_to_pad_to:
-            warnings.warn(
-                f"{max_seq_length=} is bigger than the provided {sequence_length_to_pad_to=}, overwriting the padding"
-                f" to {max_seq_length}"
-            )
-        # pad to sequence length or max seq length, whichever is bigger
-        max_seq_length = max(sequence_length_to_pad_to, max_seq_length)
+def gather_tensor(tensor, dst, group, dtype=torch.float32):
```

Comment:
there is a potential for a bunch of hangs if we don't force the dtype to be a known value(if say for whatever reason the callers called this function with different dtypes).

I fixed it here: b85bab85ed768e9dc56baaee8d35a35407a71dac  by default this function will use the input dtype(since dtype=None), but the user has the option to overwrite it